azext_iot/_params.py

```diff
@@ -119,6 +119,8 @@ def load_arguments(self, _):
         context.argument('properties', options_list=['--properties', '--props', '-p'], arg_type=event_msg_prop_type)
         context.argument('content_type', options_list=['--content-type', '--ct'],
                          help='Specify the Content-Type of the message payload to automatically format the output to that type.')
+        context.argument('device_regex', options_list=['--device-regex', '--regex'],
```

Comment:
I would prefer not to use the query technology in the name - "regex".

How about something like --device-query and support queries like we do in the "query" operation as well...
https://docs.microsoft.com/en-us/cli/azure/ext/azure-cli-iot-ext/iot/hub?view=azure-cli-latest#ext-azure-cli-iot-ext-az-iot-hub-query

Reply:
It would be great if we could support full queries as well - to allow users to query for devices or filter by tags on the devices.  I would rather implement that then this one off just for regex on device id.

@iluican

Reply:
That's a great suggestion - I'll admit in hindsight regex for device IDs is a little overkill considering wildcards pretty much handle most uses of device ID filtering. I'll edit this PR to keep it to just device ID wildcards and work on the --device-query feature in another PR

Reply:
Do you need both?  If you have --device-query, then you don't really need device wildcard.  Less surface area, less code, less bugs.

Reply:
@c-ryan-k Can you please validate if 'like' or similar wildcard matching is supported in the IoT Hub Query API. At the time of the implementation of the query command that was not supported yet. This data point will help guide the path forward.

Reply:
Correct. It doesn't not support LIKE.  

Is it possible to ask the user to use tags instead of name?  If so, then supporting the full query like in the link above would work.

If they can't, then allowing user to put a wildcard in the deviceId property and detect supported wildcards would be idea...versus a new parameter.